### ISSUE TITLE
[typescript-graphql-request] Add support for graphql-request's native batching solution.

### ISSUE THREAD

**Issue by kratam** [dotansimha/graphql-code-generator-community#204]
**Is your feature request related to a problem? Please describe.**

graphql-request supports [batching](https://github.com/prisma-labs/graphql-request#batching), but it cannot be used with the sdk.

**Describe the solution you'd like**

The plugin should generate a batched version of the queries, e.g. `sdk.queryNameBatch` where the first parameter is an array instead of an object.

**Describe alternatives you've considered**

I can create a client directly by using `GraphQLClient` from `graphql-request` and call `client.batchRequests( [ { document: QueryDocumentFromGeneratedFile } ] )` but it's not ideal (I have to manually set the return type of the method).

**Comment by akhdefault** [comment]
Can we discuss any progress on this?

**Comment by saihaj** [comment]
do you want to take this task @akhdefault ?

**Comment by akhdefault** [comment]
Yes, but I would probably need some guidance as to how it could be approached; I'll take a look around and let you know of it's clear to me how one would do this.
